handlers.py

```diff
@@ -59,69 +64,71 @@ async def registration(message: types.Message):
-    tracking_id.append(message.from_user.id)
-    tracked_username = message.text
-    try:
-        assert '@' in message.text
-    except AssertionError:
-        await bot.send_message(message.from_user.id, text="Please enter username WITH '@'")
-    result = database.tracked_id(tracked_username)
-    Person.tracked_id = result[0][0]
-    #print(tracking_id, Person.tracked_id)
-    keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
-    button1 = types.KeyboardButton("Yes", request_location=True)
-    button2 = types.KeyboardButton("No", request_contact=True)
-    keyboard.add(button1, button2)
-    await bot.send_message(Person.tracked_id,
-                           text=f"User @{message.from_user.username} wants to track you, are you agree?",
-                           reply_markup=keyboard)
+    await message.answer("Please enter phone number of a person (choose from your contacts)")
+
+    @dp.message_handler(content_types=['contact'])
+    async def find_person(message: types.Message):
+        print(message.contact['phone_number'][1::])
+        queries[database.tracked_id(message.contact['phone_number'][1::])[0][0]].append(message.from_user.id)
+        print(queries)
+        keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
+        button1 = types.KeyboardButton("Yes", request_location=True)
+        button2 = types.KeyboardButton("No")
+        keyboard.add(button1, button2)
+        await bot.send_message(database.tracked_id(message.contact['phone_number'][1::])[0][0],
+                               text=f"User {message.from_user.first_name} @{message.from_user.username} with number {database.get_contact(message.from_user.id)[0][0]} wants to track you, are you agree?",
+                               reply_markup=keyboard)
+
 
 
     @dp.message_handler(content_types=["location"])
-    async def give_position(message: types.Location):
+    async def give_position(message: types.Message):
         result = requests.get(url=f'https://geocode-maps.yandex.ru/1.x/?apikey={API_KEY}&geocode={message["location"]["longitude"]},{message["location"]["latitude"]}&format=json&lang=ru_RU')
         json_data = result.json()
-        await bot.send_message(chat_id=tracking_id[0],
-                               text=json_data['response']['GeoObjectCollection']['featureMember'][1]['GeoObject']['metaDataProperty']['GeocoderMetaData']['text'])
-        await bot.send_message(chat_id=tracking_id[0],
+        await bot.send_message(chat_id=queries[message.from_user.id][-1],
+                               text=f"User {message.from_user.first_name} @{message.from_user.username} with number {database.get_contact(message.from_user.id)[0][0]} is here:\n {json_data['response']['GeoObjectCollection']['featureMember'][1]['GeoObject']['metaDataProperty']['GeocoderMetaData']['text']}")
+        await bot.send_message(chat_id=queries[message.from_user.id][-1],
```

Comment:
again you have repeatative constructions and also `username`

Reply:
again, username is optional and not the key value to identify a user.